toolkits/slack/arcade_slack/tools/chat.py

```diff
@@ -72,43 +91,532 @@ def send_dm_to_user(
+            "current time in the format 'DD:HH:MM'"
+        ),
+    ] = None,
+    latest_relative: Annotated[
+        Optional[str],
+        (
+            "The latest message to include in the results, specified as a time offset from the "
+            "current time in the format 'DD:HH:MM'"
+        ),
+    ] = None,
+    oldest_datetime: Annotated[
+        Optional[str],
+        (
+            "The oldest message to include in the results, specified as a datetime object in the "
+            "format 'YYYY-MM-DD HH:MM:SS'"
+        ),
+    ] = None,
+    latest_datetime: Annotated[
+        Optional[str],
+        (
+            "The latest message to include in the results, specified as a datetime object in the "
+            "format 'YYYY-MM-DD HH:MM:SS'"
+        ),
+    ] = None,
+    limit: Annotated[
+        Optional[int], "The maximum number of messages to return. Defaults to 20."
+    ] = MAX_PAGINATION_LIMIT,
+    cursor: Annotated[Optional[str], "The cursor to use for pagination. Defaults to None."] = None,
+) -> Annotated[
+    dict,
+    (
+        "The conversation history / messages and next cursor for paginating results (when "
+        "there are additional messages to retrieve)."
+    ),
+]:
+    """Get the history of a conversation in Slack."""
+    conversation_metadata = await get_conversation_metadata_by_name(
+        context=context, conversation_name=conversation_name
+    )
+    return await get_conversation_history_by_id(
+        context=context,
+        conversation_id=conversation_metadata["id"],
+        oldest_relative=oldest_relative,
+        latest_relative=latest_relative,
+        oldest_datetime=oldest_datetime,
+        latest_datetime=latest_datetime,
+        limit=limit,
+        cursor=cursor,
+    )
+
+
+@tool(
+    requires_auth=Slack(
+        scopes=["channels:read", "groups:read", "im:read", "mpim:read"],
+    )
+)
+async def get_conversation_metadata_by_id(
+    context: ToolContext,
+    conversation_id: Annotated[str, "The ID of the conversation to get metadata for"],
+) -> Annotated[
+    dict,
+    (
+        "The conversation metadata. Metadata does not include: "
+        "- members, users, or people on the conversation / channel; "
+        "- history / messages of the conversation / channel."
+    ),
+]:
+    """Get the metadata of a conversation in Slack."""
+    slackClient = AsyncWebClient(token=context.authorization.token)
+
+    try:
+        response = await slackClient.conversations_info(
+            channel=conversation_id,
+            include_locale=True,
+            include_num_members=True,
         )
 
-    # Step 2: Send the message to the channel
-    response = slackClient.chat_postMessage(channel=channel_id, text=message)
-    response.validate()
+        return extract_conversation_metadata(response["channel"])
+
+    except SlackApiError as e:
+        if e.response["error"] == "channel_not_found":
```

Comment:
does the response always have an error in this case?

Reply:
In the context of a `SlackApiError` raised, yes. But I'm changing it to `.get("error")` just to be extra sure our error handling won't break in case they change their API and someone is still running an old version of our toolkit.